js/post/update_counts.js

```diff
@@ -0,0 +1,174 @@
+    'use-strict';
+
+    for (var courseCode in completed_maj) {
+        if (completed_maj.hasOwnProperty(courseCode)) {
+            if (getCookie(courseCode) === 'active' || getCookie(courseCode) === 'overridden') {
+                if (completed_maj[courseCode] < 1) {
+                    completed_maj[courseCode] += 1;
+                    if (courseCode === 'Calc1') {
+                        creditCountMaj += 1.0;
+                    } else {
+                        creditCountMaj += 0.5;
+                    }
+                } 
+            } else if ((getCookie(courseCode) === 'inactive' || getCookie(courseCode) === 'takeable')
+                        && (completed_maj[courseCode] > 0)) {
+                completed_maj[courseCode] -= 1;
+                if (courseCode === 'Calc1') {
+                    creditCountMaj -= 1.0;
+                } else {
+                    creditCountMaj -= 0.5;
+                }
+            }
+        }       
+    }
+}
+
+
+/**
+ * Updates number of completed courses in Minor.
+**/
+function updateCompletedMinCourses() {
+    'use-strict';
+
+    for (var courseCode in completed_min) {
+        if (completed_min.hasOwnProperty(courseCode)) {
+            if (getCookie(courseCode) === 'active' || getCookie(courseCode) === 'overridden') {
+                if (completed_min[courseCode] < 1) {
+                    completed_min[courseCode] += 1;
+                    creditCountMin += 0.5;
+                } 
+            } else if ((getCookie(courseCode) === 'inactive' || getCookie(courseCode) === 'takeable')
+                           && (completed_min[courseCode] > 0)) {
+                completed_min[courseCode] -= 1;
+                creditCountMin -= 0.5;
+            }
+        }       
+    }
+}
+
+
+/**
+ * Updates number of 300 level category completed courses.
+ **/
+function update300s() {
+    'use-strict';
+
+    for (var courseCode in level300) {
+        if (level300.hasOwnProperty(courseCode)) {
+            if (getCookie(courseCode) === 'active' || getCookie(courseCode) === 'overridden') {
+                if (level300[courseCode] < 1) {
+                    level300[courseCode] += 1;
+                    creditCount300 += 0.5;
+                } if ((CSCinq.indexOf(courseCode) > -1) && (activeInq.indexOf(courseCode) === -1)) { // check if Inquiry Course
+                    activeInq.push(courseCode);
+                }
+            } else if ((getCookie(courseCode) === 'inactive' || getCookie(courseCode) === 'takeable') 
+                       && (level300[courseCode] > 0)) {
+                level300[courseCode] -= 1;
+                creditCount300 -= 0.5;
+                var index = activeInq.indexOf(courseCode);
+                if (index > -1) {
+                    activeInq.splice(index, 1);
+                }
+            }
+        }       
+    }
+}
+
+
+/**
+ * Updates number of 400 level category completed courses.
+**/
+function update400s() {
```

Comment:
This function seems to be repeated, it might be possible to refactor it and pass in level400/300 as a parameter.